pre-check/v1.x/check.sh

```diff
@@ -264,8 +268,17 @@ check_volumes()
                     rm -f $healthy_state
                 else
                     robustness=$(echo $volume_json | jq -r '.status.robustness')
+                    state=$(echo $volume_json | jq -r '.status.state')
                     if [ "$robustness" = "healthy" ]; then
                         log_verbose "Volume ${lh_volume} is healthy."
+                    elif [ "$state" = "detached" ]; then
+                        replica_count=$(kubectl get replicas -n longhorn-system -l longhornvolume=$lh_volume -o json | jq '.items | length')
+                        if [ $replica_count -le 1 ]; then
+                            log_info "Degraded Longhorn Volume found: ${lh_volume}"
+                            rm -f $healthy_state
+                        else
+                            log_info "Detached Longhorn Volume found: ${lh_volume}"
+                        fi
```

Comment:
Suggest to compare volume's `.spec.numberOfReplicas` with the existing replica CR:

```suggestion
                    elif [ "$state" = "detached" ]; then
                        replica_count=$(kubectl get replicas -n longhorn-system -l longhornvolume=$lh_volume -o json | jq '.items | length')
                        if [ $replica_count -lt $volume_replicas ]; then
                            log_info "Detached Longhorn Volume: ${lh_volume} w/o enough replicas"
                            rm -f $healthy_state
                        else
                            log_info "Detached Longhorn Volume found: ${lh_volume}"
                        fi
```